lib/game.rb

```diff
@@ -0,0 +1,101 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [[0,1,2], [3,4,5], [6,7,8],
+                      [0,3,6], [1,4,7], [2,5,8],
+                      [0,4,8], [6,4,2]]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board=board
+  end
+
+  def board
+    @board
+  end
+
+  def player_1
+    @player_1
+  end
+
+  def player_2
+    @player_2
+  end
+
+  def current_player
+    if (@board.cells.find_all {|t| t == "X" || t=="O"}).size % 2 == 0
+      return @player_1
+    else
+      return @player_2
+    end
+  end
+
+  def over?
+    #binding.pry
+    #if @board.cells.detect {|space| space == " "}
+    #  return false
+    #else
+    draw? || won? #
+    #end # true
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      @board.cells[combo[0]] == @board.cells[combo[1]]  &&
+      @board.cells[combo[1]] == @board.cells[combo[2]] &&
+      @board.taken?(combo[0]+1)
+    end
+    #binding.pry
+  end
+
+  def draw?
+    @board.full? && !won?
+=begin
+    WIN_COMBINATIONS.each do |combo|
+      if @board.cells[combo[0]] == @board.cells[combo[1]]  &&  @board.cells[combo[1]] == @board.cells[combo[2]]
+        return false
+      end
+    end
+    return true
+=end
+  end
+
+  def winner
```

Comment:
You can use the `won?` method here 
```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.